generators/php/codegen/src/context/AbstractPhpGeneratorContext.ts

```diff
@@ -218,6 +245,115 @@ export abstract class AbstractPhpGeneratorContext<
         }
     }
 
+    public dereferenceOptional(typeReference: TypeReference): TypeReference {
+        if (typeReference.type === "container" && typeReference.container.type === "optional") {
+            return typeReference.container.optional._visit({
+                container: (container) => {
+                    return this.dereferenceOptional(TypeReference.container(container));
+                },
+                named: (named) => {
+                    return this.dereferenceOptional(TypeReference.named(named));
+                },
+                primitive: (primitive) => {
+                    return TypeReference.primitive(primitive) as TypeReference;
+                },
+                unknown: () => {
+                    return TypeReference.unknown() as TypeReference;
+                },
+                _other: () => {
+                    return TypeReference.unknown() as TypeReference;
+                }
+            });
+        }
+
+        if (typeReference.type === "named") {
+            const typeDeclaration = this.getTypeDeclarationOrThrow(typeReference.typeId);
+            if (typeDeclaration.shape.type === "alias") {
+                return typeDeclaration.shape.aliasOf;
+            }
+        }
+
+        // Original type was not optional
+        return typeReference;
+    }
+
+    public dereferenceCollection(typeReference: TypeReference): TypeReference {
```

Comment:
Similar to above w.r.t similarity with the `isOptional` helper.

Reply:
done